Java/AutoML/automl-cms-api/src/main/java/com/automl/cms/api/controller/MLDatasetController.java

```diff
@@ -0,0 +1,54 @@
+package com.automl.cms.api.controller;
+
+import static com.automl.cms.api.model.Result.ok;
+
+import java.util.List;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.automl.component.service.MLDatasetOperatorService;
+import com.automl.component.vo.MLDatasetVO;
+
+/**
+ * @author wangqingwei
+ * Created on 2022-08-10
+ */
+@RestController
+@RequestMapping("/namespaces/{namespace}/datasets")
+public class MLDatasetController {
+    private static final Logger logger = LoggerFactory.getLogger(MLDatasetController.class);
+
+    @Autowired
+    private MLDatasetOperatorService mlDatasetOperatorService;
+
+    @PostMapping(value = "/")
+    public Object putDataset(@PathVariable String namespace,
+        @RequestParam(value = "name", defaultValue = "") String name,
+        @RequestParam(value = "description", defaultValue = "") String description,
+        @RequestParam(value = "displayName", defaultValue = "") String displayName,
+        @RequestParam(value = "uri", defaultValue = "") String uri) {
+        logger
+            .info("putDataset req namespace : {}, name : {}, description : {}, displayName : {}, uri : {}", namespace, name, description,
+                displayName, uri);
+        MLDatasetVO mlDatasetVO =
+            mlDatasetOperatorService.insertDataset(namespace, name, description, displayName, uri);
+        return ok(mlDatasetVO);
+    }
+
+    @GetMapping(value = "/")
+    public Object getDatasetList(@PathVariable String namespace,
+        @RequestParam(value = "offset", required = false, defaultValue = "0") int offset,
+        @RequestParam(value = "limit", required = false, defaultValue = "100") int limit) {
+        final List<MLDatasetVO> datasetVOList =
+            mlDatasetOperatorService.getByNamespaceAndLimit(namespace, offset, limit);
+        return ok(datasetVOList);
```

Comment:
It looks like the response is inconsistent with the following operationId described in OpenAPI doc.
automl.namespaces.datasets.list

Reply:
you can saw this class from com.automl.cms.api.model.Result.
sorry, I see that /namespaces/{namespace}/datasets get Response has only code and a JSON.
How does the front-end developer end get this JSON data,  so i add column 'msg' to json mark, 
the front-end developer can  use 'msg' to  get data. 
I'm not familiar with this OpenAPI, so the markup that JSON returns to the front-end developer is using content or others ?

Reply:
For the yaml file, you can check OpenAPI protocol to know more about it. Or you can use the swagger editor to visualize it like the screenshot below.
![image](https://user-images.githubusercontent.com/98159352/184500218-a604f265-7537-4806-abee-0a05ed629db0.png)

Reply:
Please notice my new update, resources list will be put into the value of key results, in case that an implementation of pagination may also provide the total count of items in an int32 field named total_size

![image](https://user-images.githubusercontent.com/98159352/184501913-660e2a2e-9abb-4e16-9064-7a737bf66138.png)

Reply:
ok, I find this description from https://openapi.apifox.cn/#response-%E5%AF%B9%E8%B1%A1 
i will change msg to content,  In addition, Whether to query the namespace or dataset in the path with ID instead of name

Reply:
ok, I write new commit [[add global-exception-handler and controller resp]](https://github.com/ml-for-good/automl/pull/71/commits/78397ffdaeed7e757c1f3207cb5fcb4f56714070) to deal with some problem.

Reply:
content is not a key in the rest response, content is used to describe the response. In the above screenshot, results is the key in the rest response

Reply:
saw this interface,  this interface resp key is  ? 
<img width="867" alt="截屏2022-08-14 14 10 42" src="https://user-images.githubusercontent.com/48853704/184524839-9c1e973b-fed9-432c-b331-b1d7106c5d4f.png">
<img width="860" alt="截屏2022-08-14 14 10 49" src="https://user-images.githubusercontent.com/48853704/184524845-f9bd0f61-08f0-44f0-b7a5-1e231a89156a.png">

all interface resp should has oneself key,  the front-end developer can use 'content' to get data.

Reply:
In the OpenAPI definition, name is an alias of ID. Which one do you prefer? Do you want me to replace all name with ID?

Reply:
Use ID directly, as long as the page displays name

Reply:
I've replaced name with id, and also replaced displayName with name in some cases.

Reply:
ok, i see.  The requested parameter type in all interface is a long(int64). thanks!    i will update refer table design. 
<img width="881" alt="截屏2022-08-14 15 32 19" src="https://user-images.githubusercontent.com/48853704/184527076-af55503a-6722-4b4b-95e6-f982cae2ca69.png">
<img width="858" alt="截屏2022-08-14 15 33 33" src="https://user-images.githubusercontent.com/48853704/184527086-11c91634-5c7c-4764-9d4b-ac5edca64eee.png">

case: 
CREATE TABLE `ml_dataset`
(
    `id`           bigint(20) unsigned AUTO_INCREMENT NOT NULL COMMENT 'id',
)

Reply:
There is also a POST interface /namespaces/{namespace}/datasets

Namespace in the returned value refers to the name of the space to which the dataset belongs, right? In this case, the request parameter is not recommended to be reused #/components/schemas/Dataset .

You are advised to use a single parameter: name, description, URI.

The namespaceId can be obtained from the interface URL and will be automatically generated by the createTime system.



If it returns an ID, you can reuse it

Reply:
Good point! I changed namespace to namespaceId, and marked it as Output only.

Reply:
ok, i saw you commit.  
i write new commit [[deal with error case]](https://github.com/ml-for-good/automl/pull/71/commits/59fd051107aee4402c9df385faa805f7bc03272c) to deal with error case.

Reply:
> int64
I also updated the data format of id.

Reply:
Ok, so my submission has the same ID and namespaceId as yours, both int64.

Reply:
> Ok, so my submission has the same ID and namespaceId as yours, both int64.

Unfortunately, the browser doesn't support int64 by default, maybe the backend need to transform all long data type fields into string. What's your idea?
Please check this article https://blog.csdn.net/doctor_who2004/article/details/104026152

Reply:
ok, I fixed from [[fix Java readme.md and front-end Long to string]](https://github.com/ml-for-good/automl/pull/71/commits/3492754257a61a7581358733cb279ea398a22c01)